webapp/channels/src/components/desktop_auth_token.tsx

```diff
@@ -0,0 +1,240 @@
+    };
+
+    const openDesktopApp = () => {
+        if (!SiteURL) {
+            return;
+        }
+        const url = new URL(SiteURL);
+        const redirectTo = query.get('redirect_to');
+        if (redirectTo) {
+            url.pathname += redirectTo;
+        }
+        url.protocol = 'mattermost';
+        window.location.href = url.toString();
+    };
+
+    useEffect(() => {
+        setShowBottomMessage(false);
+
+        const timeout = setTimeout(() => {
+            setShowBottomMessage(true);
+        }, BOTTOM_MESSAGE_TIMEOUT) as unknown as number;
+
+        return () => {
+            clearTimeout(timeout);
+        };
+    }, [status]);
+
+    useEffect(() => {
+        if (!token) {
+            return () => {};
+        }
+
+        const url = getExternalLoginURL();
+        window.open(url);
+
+        setStatus(DesktopAuthStatus.Polling);
+        interval.current = setInterval(tryDesktopLogin, POLLING_INTERVAL);
+
+        return () => {
+            clearInterval(interval.current as unknown as number);
+        };
+    }, [token]);
+
+    useEffect(() => {
+        if (status === DesktopAuthStatus.Complete) {
+            openDesktopApp();
+            return;
+        }
+
+        setToken(crypto.randomBytes(32).toString('hex'));
+    }, []);
+
+    let mainMessage;
+    let subMessage;
+    let bottomMessage;
+
+    if (status === DesktopAuthStatus.Polling) {
+        mainMessage = (
+            <FormattedMessage
+                id='desktop_auth_token.polling.redirectingToBrowser'
+                defaultMessage='Redirecting to browser...'
+            />
+        );
+        subMessage = (
+            <FormattedMessage
+                id='desktop_auth_token.polling.awaitingToken'
+                defaultMessage='Authenticating in the browser, awaiting valid token.'
+            />
+        );
+
+        bottomMessage = (
+            <FormattedMessage
+                id='desktop_auth_token.polling.isComplete'
+                defaultMessage='Authentication complete? <a>Check token now</a>'
+                values={{
+                    a: (chunks: React.ReactNode) => {
+                        return (
+                            <a onClick={tryDesktopLogin}>
+                                {chunks}
+                            </a>
+                        );
+                    },
+                }}
+            />
+        );
+    }
+
+    if (status === DesktopAuthStatus.Complete) {
+        mainMessage = (
+            <FormattedMessage
+                id='desktop_auth_token.complete.youAreNowLoggedIn'
+                defaultMessage='You are now logged in'
+            />
+        );
+        subMessage = (
+            <FormattedMessage
+                id='desktop_auth_token.complete.openMattermost'
+                defaultMessage='Click on <b>Open Mattermost</b> in the browser prompt to <a>launch the desktop app</a>'
```

Comment:
Minor nit, but I'd tend towards writing out `link` instead of `a` for these so that it's clearer to the translator what that is. While they're both HTML, `a` is less intuitive to people who aren't familiar with it, and something about seeing `a` and `b` together in this message made me feel like we're naming our variables a, b, c, etc when I first read this